packages/react-components/src/components/time-sync/timeSelection.tsx

```diff
@@ -52,31 +55,93 @@ const messages: ViewportMessages = {
  * parent TimeSync and affect all other viewports in
  * that TimeSync group.
  */
-export const TimeSelection = () => {
+export const TimeSelection = ({ isPaginationEnabled }: { isPaginationEnabled?: boolean }) => {
   const { viewport, setViewport } = useViewport();
+  const [disabled, setDisabled] = useState<boolean | undefined>(true);
 
   const handleChangeDateRange: NonCancelableEventHandler<DateRangePickerProps.ChangeDetail> = (event) => {
     const { value } = event.detail;
-
     if (!value) return;
-
+    if (value.type === 'relative') setDisabled(true);
     setViewport(dateRangeToViewport(value), 'date-picker');
   };
 
+  const handlePaginateForward = () => {
+    if (!viewport) return;
+    const value = viewportToDateRange(viewport);
+    if (value?.type == 'absolute' && !disabled) {
+      const duration = new Date(value.endDate).getTime() - new Date(value.startDate).getTime();
+      const newEnd = new Date(new Date(value.endDate).getTime() + duration);
+      if (newEnd >= new Date()) {
+        setDisabled(true)
+      } else {
```

Comment:
It's not clear to me why the button can be clicked and then become disabled without any mutation of state occurring.